OracleDatabase/SingleInstance/dockerfiles/18.4.0/runOracle.sh

```diff
@@ -0,0 +1,218 @@
+#!/bin/bash
+
+############# Execute custom scripts ##############
+function runUserScripts {
+
+  SCRIPTS_ROOT="$1";
+
+  # Check whether parameter has been passed on
+  if [ -z "$SCRIPTS_ROOT" ]; then
+    echo "$0: No SCRIPTS_ROOT passed on, no scripts will be run";
+    exit 1;
+  fi;
+  
+  # Execute custom provided files (only if directory exists and has files in it)
+  if [ -d "$SCRIPTS_ROOT" ] && [ -n "$(ls -A $SCRIPTS_ROOT)" ]; then
+      
+    echo "";
+    echo "Executing user defined scripts"
+  
+    for f in $SCRIPTS_ROOT/*; do
+        case "$f" in
+            *.sh)     echo "$0: running $f"; . "$f" ;;
+            *.sql)    echo "$0: running $f"; echo "exit" | su -p oracle -c "$ORACLE_HOME/bin/sqlplus / as sysdba @$f"; echo ;;
+            *)        echo "$0: ignoring $f" ;;
+        esac
+        echo "";
+    done
+    
+    echo "DONE: Executing user defined scripts"
+    echo "";
+  
+  fi;
+  
+}
+
+########### Move DB files ############
+function moveFiles {
+   if [ ! -d $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID ]; then
+      su -p oracle -c "mkdir -p $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/"
+   fi;
+   
+   su -p oracle -c "mv $ORACLE_HOME/dbs/spfile$ORACLE_SID.ora $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/"
+   su -p oracle -c "mv $ORACLE_HOME/dbs/orapw$ORACLE_SID $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/"
+   su -p oracle -c "mv $ORACLE_HOME/network/admin/listener.ora $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/"
+   su -p oracle -c "mv $ORACLE_HOME/network/admin/tnsnames.ora $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/"
+   mv /etc/sysconfig/oracle-xe-18c.conf $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/
```

Comment:
What's the reason for preserving the `oracle-xe-18c.conf` file? It's only used for DB creation which is done only at the beginning of the container start.

Reply:
I don't remember why I am added it, also re-creating a container with other parameters throws errors like:

> ln: failed to create symbolic link '/etc/sysconfig/oracle-xe-18c.conf': File exists
#########################
DATABASE IS READY TO USE!
#########################

so its OK to remove this symbolic link from the script.